AdvancedVehicleOptions/GUI/UIOptionPanel.cs

```diff
@@ -731,7 +731,7 @@ protected void OnCapacitySubmitted(UIComponent component, string text)
 
             m_options.capacity = int.Parse(text);
             VehicleOptions.prefabUpdateUnits = m_options.prefab;
-            new EnumerableActionThread(VehicleOptions.UpdateCapacityUnits);
+            SimulationManager.instance.AddAction(VehicleOptions.UpdateCapacityUnits);

```

Comment:
This one and also that in line `746` are my main concerns.
We are performing async operation (same as before change) but you are immediately setting `m_initialized` field. 
That won't work as expected when simulation is running and the user has "fast fingers" (could click multiple times).

Setting static field in the line `733` and then triggering an asynchronous action which reads that file later is very bad idea. With fast fingers or when task is slow it is possible to switch prefab while first task is still processing previous prefab. I'm sure you get the idea what can happen😉